libbeat/outputs/kafka/config.go

```diff
@@ -307,3 +319,21 @@ func newSaramaConfig(log *logp.Logger, config *kafkaConfig) (*sarama.Config, err
 	}
 	return k, nil
 }
+
+// makeBackoffFunc returns a stateless implementation of exponential-backoff-with-jitter. It is conceptually
+// equivalent to the stateful implementation used by other outputs, EqualJitterBackoff.
+func makeBackoffFunc(cfg backoffConfig) func(retries, maxRetries int) time.Duration {
+	maxBackoffRetries := int(math.Ceil(math.Log2(float64(cfg.Max) / float64(cfg.Init))))
+
+	return func(retries, _ int) time.Duration {
+		temp := (uint64(cfg.Init) * uint64(1<<retries)) / 2
+		jitter := uint64(rand.Int63n(int64(temp)))
+
+		backoff := time.Duration(temp + jitter)
+		if retries >= maxBackoffRetries {
+			backoff = cfg.Max
+		}
+
+		return backoff
+	}
```

Comment:
checking the implementation for EqualJitterBackoff, we always apply jitter, even after hitting max.

maybe try this:
```
func(retries, _ int) time.Duration {
  // compute 'base' duration for exponential backoff
  var dur := cfg.Max
  if retries < maxBackoffRetries {
    dur = time.Duration((uint64(cfg.Init) * uint64(1<<retries)))
  }

  // apply about equaly distributed jitter in second half of the interval, such that the wait time falls into the interval [dur/2, dur]
  limit := int64(dur / 2)
  jitter := rand.Int63n(limit + 1)
  return time.Duration(limit + jitter)
}
```

Reply:
Ah yes, you're right, 🤦. Will fix!

Reply:
Fixed in 5eb1cbd13cb02ae25978fa3270e6cf36207d382d.

Reply:
Let's add unit tests for the backoff function :)

Reply:
I added a unit test but please take a look. It doesn't make very strong assertions about what the backoff values after each call to the backoff function should be. I'm not sure if/how we can make even stronger assertions that the test is making right now.